servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/SetDynamicCompositeCancellable.java

```diff
@@ -73,7 +80,9 @@ public boolean add(Cancellable toAdd) {
         if (!cancellables.add(toAdd)) {
             toAdd.cancel(); // out of memory, or user has implemented equals/hashCode so there is overlap.
         } else if (isCancelled()) {
-            cancelAll(); // avoid concurrency on toAdd if another thread invokes cancel(), just cancel all.
```

Comment:
note this comment. I don't think we can safely remove/cancel here because `Set` doesn't provide an API to "atomically" get/remove items during iteration (like `Queue#poll()`). This change may lead to concurrent invocation of `toAdd.cancel()`.

Reply:
`NavigableSet` provides [pollFirst](https://docs.oracle.com/javase/7/docs/api/java/util/NavigableSet.html#pollFirst()) and [ConcurrentSkipListSet](https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ConcurrentSkipListSet.html) is an implementation we could use, but this requires item ordering (e.g. `Comparable`, or `Comparator`) which doesn't seem practical in this case. We can always punt on the `SetDynamicCompositeCancellable` specific part of this change.